Src/Plugins/Security/SecurePlaintextSecretsValidators/SEC101_033.MongoDbCredentialsValidator.cs

```diff
@@ -1,23 +1,26 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
 using System.Collections.Generic;
+using System.Linq;
 
 using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk;
 using Microsoft.RE2.Managed;
 
+using MongoDB.Driver;
+
 namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
 {
-    public class MongoDbCredentialsValidator : StaticValidatorBase
+    public class MongoDbCredentialsValidator : DynamicValidatorBase
     {
         protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary<string, FlexMatch> groups)
         {
-            if (!groups.TryGetNonEmptyValue("id", out FlexMatch id) ||
-                !groups.TryGetNonEmptyValue("host", out FlexMatch host) ||
-                !groups.TryGetNonEmptyValue("secret", out FlexMatch secret))
-            {
-                return ValidationResult.CreateNoMatch();
-            }
+            groups.TryGetValue("id", out FlexMatch id);
```

Comment:
>TryGetValue [](http://example.com/codeflow?start=19&length=11)

why are you using try value? your regexes seem to require some of this data?

Reply:
"mongodb+srv://" format is used to connect to a DNS host to resolve multiple host names. Its not compatible with the standard format "mongodb://".

https://www.mongodb.com/docs/manual/reference/connection-string/#srv-connection-format